hw_diag/tests/test_views_diagnostics.py

```diff
@@ -86,3 +87,27 @@ def test_version_endpoint(self):
         self.assertEqual(reply['firmware_version'], '1337.13.37')
         self.assertEqual(reply['diagnostics_version'], 'aabbffe')
         self.assertEqual(reply['firmware_short_hash'], '0011223')
+
+    @patch('hw_diag.diagnostics.add_gateway_txn_diagnostic.AddGatewayTxnDiagnostic.perform_test')
+    def test_add_gateway_txn_success(self, mock):
+        endpoint = '/v1/add-gateway-txn'
+        data = dict(file=(BytesIO(), 'shipping_destination.json.asc'),)
+        resp = self.client.post(endpoint, content_type='multipart/form-data', data=data)
+
+        self.assertEqual(resp.status_code, 200)
+        self.assertDictEqual(resp.json, {
+            'diagnostics_passed': True,
+            'errors': []
+        })
+
+    @patch('hw_diag.diagnostics.add_gateway_txn_diagnostic.AddGatewayTxnDiagnostic.perform_test')
+    def test_add_gateway_txn_failure_no_file_payload(self, mock):
+        endpoint = '/v1/add-gateway-txn'
+        resp = self.client.post(endpoint, data=None)
+
+        self.assertEqual(resp.status_code, 406)
+        self.assertDictEqual(resp.json, {
+            'diagnostics_passed': False,
+            'errors': ['error_message', 'error_message'],
```

Comment:
I find this kind of confusing. The keys normally indicate some kind of scope about the information, but `error_message` is confusing out of context. I think the key should still be `DESTINATION_ADD_GATEWAY_TXN_KEY`.

Reply:
Make sense. Ok, I am going to use (key, err_msg) as the parameter.